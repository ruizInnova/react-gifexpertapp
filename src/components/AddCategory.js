import React, { useState } from 'react'
//validation
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e) => { 
        console.log( e.target.value); 
        setinputValue( e.target.value );
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2) {

            setCategories(cats => [inputValue, ...cats ] );
            setinputValue('');
            console.log('submit realizado..');
        }
    };

    return (
        <form onSubmit={ handleSubmit }>
            <h4>{inputValue}</h4>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

//Add propTypes required 

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
