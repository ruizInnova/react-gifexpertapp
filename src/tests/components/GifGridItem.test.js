import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Test for component <GifGridItem />', ()=>{
    test('must display the information correctly', ()=>{

        const wrapper = shallow( <GifGridItem />  );
        expect( wrapper ).toMatchSnapshot();

    })
});