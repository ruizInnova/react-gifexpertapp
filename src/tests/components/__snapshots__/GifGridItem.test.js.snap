// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for component <GifGridItem /> must display the information correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <h4 />
</div>
`;
